src/watch.rs

```diff
@@ -25,11 +25,13 @@ pub struct WatchSystem {
     build_rx: Receiver<PathBuf>,
     /// The watch system used for watching the filesystem.
     _watcher: (JoinHandle<()>, RecommendedWatcher),
+    /// Sender to signal when a build has finished and the page can be refreshed
+    build_done_tx: Option<Sender<()>>,
```

Comment:
For now, this is good. Long-term we will probably want to update the injected code to include a light UI which is able to describe the error directly in a UI overlay. At that point in time, we can update this signal to carry additional info on any build related errors.